client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/unified-design-picker.tsx

```diff
@@ -0,0 +1,390 @@
+						? siteVerticalId
+						: ''
+				)
+			);
+			recordTracksEvent( 'calypso_signup_select_design', {
+				...getEventPropsByDesign( _selectedDesign ),
+				...( positionIndex >= 0 && { position_index: positionIndex } ),
+			} );
+
+			handleSubmit( {
+				selectedDesign: _selectedDesign,
+				selectedSiteCategory: categorization.selection,
+			} );
+		}
+	}
+
+	function previewDesign( _selectedDesign: Design, positionIndex?: number ) {
+		recordTracksEvent( 'calypso_signup_design_preview_select', {
+			...getEventPropsByDesign( _selectedDesign ),
+			...( positionIndex && { position_index: positionIndex } ),
+		} );
+
+		setSelectedDesign( _selectedDesign );
+		setIsPreviewingDesign( true );
+	}
+
+	function upgradePlan() {
+		if ( ! isEnabled( 'signup/seller-upgrade-modal' ) ) {
+			return goToCheckout();
+		}
+
+		recordTracksEvent( 'calypso_signup_design_upgrade_modal_show', {
+			theme: selectedDesign?.slug,
+		} );
+		setShowUpgradeModal( true );
+	}
+
+	function goToCheckout() {
+		if ( ! isEnabled( 'signup/design-picker-premium-themes-checkout' ) ) {
+			return null;
+		}
+
+		if ( isEnabled( 'signup/seller-upgrade-modal' ) ) {
+			recordTracksEvent( 'calypso_signup_design_upgrade_modal_checkout_button_click', {
+				theme: selectedDesign?.slug,
+			} );
+		}
+
+		const plan = isEligibleForProPlan && isEnabled( 'plans/pro-plan' ) ? 'pro' : 'premium';
+
+		if ( siteSlugOrId ) {
+			const params = new URLSearchParams();
+			params.append( 'redirect_to', window.location.href.replace( window.location.origin, '' ) );
+
+			// The theme upsell link does not work with siteId and requires a siteSlug.
+			// See https://github.com/Automattic/wp-calypso/pull/64899
+			window.location.href = `/checkout/${ encodeURIComponent(
+				siteSlug || urlToSlug( site?.URL || '' ) || ''
+			) }/${ plan }?${ params.toString() }`;
+		}
+	}
+
+	function recordStepContainerTracksEvent( eventName: string ) {
+		const tracksProps = {
+			step: 'design-step',
+			intent: intent,
+		};
+
+		recordTracksEvent( eventName, tracksProps );
+	}
+
+	const handleBackClick = () => {
+		if ( isPreviewingDesign ) {
+			setSelectedDesign( undefined );
+			setIsPreviewingDesign( false );
+			return;
+		}
+
+		goBack();
+	};
+
+	// Track scroll event to make sure people are scrolling on mobile.
+	useTrackScrollPageFromTop( isMobile && ! isPreviewingDesign, flow || '', STEP_NAME );
+
+	// Make sure people is at the top when entering/leaving preview mode.
+	useEffect( () => {
+		window.scrollTo( { top: 0 } );
+	}, [ isPreviewingDesign ] );
+
+	// Don't render until we've fetched the designs from the backend.
+	if ( ! site || isLoadingDesigns ) {
+		return null;
+	}
+
+	if ( selectedDesign && isPreviewingDesign ) {
+		let designTitle;
+		if ( selectedDesign.design_type === 'vertical' ) {
+			const designIndex = generatedDesigns.findIndex(
+				( design ) => design.slug === selectedDesign.slug
+			);
+			designTitle = translate( 'Generated design option #%(designNumber)s', {
```

Comment:
cc: @taipeicoder do we want it blank or `Generated design option #1`?

Anyway, we actually never say `Generated` to the users... maybe replace it with `Custom`? As the heading is `Custom designs for your site` 😄

Reply:
Actually I prefer to be empty like the design:
![Screen Shot 2022-07-25 at 11 27 35 PM](https://user-images.githubusercontent.com/797888/180816495-07fcc1ed-1ccd-4348-bd9e-6bf1462b9c4b.png)

The "design option #1" suggestion was for the thumbnail's aria-label 😅.
I think if we display "Generated/custom design option #1" in the preview header, it'd be too placeholder-y.

Reply:
ahh this is a good point, I should have gone back and checked the design!

Reply:
Looks good to me now 😄 
<img width="1231" alt="image" src="https://user-images.githubusercontent.com/1525580/180915917-3f2edd40-ebb4-4771-803e-02c52deb02e9.png">